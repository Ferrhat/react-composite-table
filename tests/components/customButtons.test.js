import React from 'react';
import { shallow } from 'enzyme';
import CustomButtons from '../../lib/components/customButtons';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('CustomButtons',()=>{
    const mockButtonHandler = jest.fn();
    let button = shallow(<CustomButtons buttons={[
        {
            icon: { type: 'icon', name: 'edit' },
            handler: mockButtonHandler,
        },
        {
            renderer: () => <h1>{'testH1'}</h1>,
        }
    ]} row={{id: 1}} />);

    it('renders properly', () => {
        expect(button).toMatchSnapshot();
    });

    it('calls the onclick function', () => {
        button = shallow(<CustomButtons buttons={[
            {
                icon: { type: 'icon', name: 'edit' },
                handler: mockButtonHandler,
            },
            {
                renderer: () => <h1>{'testH1'}</h1>,
            }
        ]} row={{id: 1}} />);
        expect(button.find('li > span.icon-edit').length).toEqual(1);
        button.find('li > span.icon-edit').simulate('click');
        expect(mockButtonHandler).toBeCalled();
        expect(button.find('li > h1').length).toEqual(1);
        expect(button.find('li > h1').text()).toEqual('testH1');
    });
});
