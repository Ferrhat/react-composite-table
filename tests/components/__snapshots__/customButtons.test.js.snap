// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButtons renders properly 1`] = `
<React.Fragment>
  <li
    className="list-inline-item"
    key="1-0"
  >
    <span
      className="icon icon-edit"
      onClick={[Function]}
    >
       
       
    </span>
  </li>
  <li
    className="list-inline-item"
    key="1-1"
  >
    <h1>
      testH1
    </h1>
  </li>
</React.Fragment>
`;
